src/middleware.ts

```diff
@@ -122,13 +122,52 @@ type StateStorage = {
   setItem: (name: string, value: string) => void | Promise<void>
 }
 type PersistOptions<S> = {
+  /** Name of the storage (must be unique) */
   name: string
-  storage?: StateStorage
+  /**
+   * A function returning a storage.
+   * The storage must fit `window.localStorage`'s api.
+   * For example the storage could be `AsyncStorage` from React Native.
+   *
+   * @default () => localStorage
+   */
+  storage?: () => StateStorage
```

Comment:
```suggestion
  getStorage?: () => StateStorage
```
We could name like that. Not a strong opinion, but more precise and less breaking change.